modules/blocks.php

```diff
@@ -190,3 +190,72 @@ function jetpack_slideshow_block_load_assets( $attr, $content ) {
 	Jetpack_Gutenberg::load_assets_as_required( 'slideshow', $dependencies );
 	return $content;
 }
+
+/**
+ * Mailchimp Block.
+ */
+jetpack_register_block(
+	'mailchimp',
+	array(
+		'render_callback' => 'jetpack_mailchimp_block_load_assets',
+	)
+);
+
+/**
+ * Mailchimp block registration/dependency declaration.
+ *
+ * @param array $attr - Array containing the map block attributes.
+ *
+ * @return string
+ */
+function jetpack_mailchimp_block_load_assets( $attr ) {
+	$values  = array();
+	$blog_id = ( defined( 'IS_WPCOM' ) && IS_WPCOM ) ?
+		get_current_blog_id() : Jetpack_Options::get_option( 'id' );
+	Jetpack_Gutenberg::load_assets_as_required( 'mailchimp', null );
+	$defaults = array(
+		'title'            => esc_html__( 'Join my email list', 'jetpack' ),
+		'emailPlaceholder' => esc_html__( 'Enter your email', 'jetpack' ),
+		'submitLabel'      => esc_html__( 'Join My Email List', 'jetpack' ),
+		'consentText'      => esc_html__( 'By clicking submit, you agree to share your email address with the site owner and MailChimp to receive marketing, updates, and other emails from the site owner. Use the unsubscribe link in those emails to opt out at any time.', 'jetpack' ),
+		'processingLabel'  => esc_html__( 'Processing…', 'jetpack' ),
+		'successLabel'     => esc_html__( 'Success! You\'ve been added to the list.', 'jetpack' ),
+		'errorLabel'       => esc_html__( 'Oh no! Unfortunately there was an error. Please try reloading this page and adding your email once more.', 'jetpack' ),
+	);
+	foreach ( $defaults as $id => $default ) {
+		$values[ $id ] = isset( $attr[ $id ] ) ? $attr[ $id ] : $default;
+	}
+	ob_start();
+	?>
+	<div class="wp-block-jetpack-mailchimp" data-blog-id="<?php echo esc_attr( $blog_id ); ?>">
```

Comment:
I would recommend adding the classes that may come from `$attr['className']` in here as well.